core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -26,6 +38,9 @@
     private final FeatureSupplier featureSupplier;
     private final Plugins plugins;
 
+    //TODO: would be nice to make this private and enforce usage of Builder
+    //TODO: this would mean any changes required would be nicely hidden away
+    //TODO: and all the complexity of setup remains within the builder itself
```

Comment:
While it hides a bunch of boiler plate code, this isn't real complexity.

Reply:
The sensible defaults are really nice though.

Reply:
OK, I just generally go with the tactic of hiding construction of complex classes behind builders - not a problem not hiding though

Reply:
What keeps you from making this private?

Reply:
Within this repo - nothing

Just wasn't sure of any other cucumber projects utilising `Runtime`s constructor and wanted to play it safe

Reply:
Then we can make it private. 

- Breaking changes are allowed on majors.
- I already broke the whole Runtime with my re-factoring
- We're only applying semantic versioning to the api packages anyway.